client/src/components/Register/hooks/useRegister.tsx

```diff
@@ -0,0 +1,56 @@
+import { useQueryClient } from "@tanstack/react-query";
+import { ChangeEvent, useCallback, useState } from "react";
+import useRegisterMutation from "../../../helpers/fetch/user/useRegisterMutation";
+import { isKeyofNewUser, isValidNewUser } from "../helpers/validate";
+import { NewUser } from "../types/new-user.types";
+
+const defaultNewUser: NewUser = {};
+
+export default function useRegister() {
+	const [newUser, setNewUser] = useState<NewUser>(defaultNewUser);
+	const [message, setMessage] = useState<string | null>();
+	const { mutate } = useRegisterMutation();
+	const client = useQueryClient();
+
+	/** Update newUser using input's name and value. */
+	function handleChange(e: ChangeEvent<HTMLInputElement>) {
+		const { name, value } = e.currentTarget; // `name` should be keyof NewUser
```

Comment:
Could do without the comments since the `isKeyofNewUser` check happens on the next line.